app/components/search_bar/search_bar.android.js

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+import React, {PropTypes, PureComponent} from 'react';
+import {
+    Keyboard,
+    TextInput,
+    StyleSheet,
+    View,
+    TouchableOpacity
+} from 'react-native';
+import Icon from 'react-native-vector-icons/MaterialIcons';
+
+const styles = StyleSheet.create({
+    searchBar: {
+        flexDirection: 'row',
+        alignItems: 'center',
+        backgroundColor: 'transparent'
+    },
+    searchBarInput: {
+        flex: 1,
+        fontWeight: 'normal',
+        backgroundColor: 'transparent'
+    }
+});
+
+export default class SearchBarAndroid extends PureComponent {
+    static propTypes = {
+        height: PropTypes.number.isRequired,
+        fontSize: PropTypes.number.isRequired,
+        text: PropTypes.string,
+        placeholder: PropTypes.string,
+        showCancelButton: PropTypes.bool,
+        textFieldBackgroundColor: PropTypes.string,
+        placeholderTextColor: PropTypes.string,
+        textColor: PropTypes.string,
+        onChange: PropTypes.func,
+        onChangeText: PropTypes.func,
+        onFocus: PropTypes.func,
+        onBlur: PropTypes.func,
+        onSearchButtonPress: PropTypes.func,
+        onCancelButtonPress: PropTypes.func
+    };
+
+    static defaultProps = {
+        placeholder: 'Search',
+        showCancelButton: true,
+        placeholderTextColor: '#bdbdbd',
+        textColor: '#212121'
+    };
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            isOnFocus: false,
+            value: this.props.text
+        };
+        this.onFocus = this.onFocus.bind(this);
+        this.onBlur = this.onBlur.bind(this);
+        this.onCancelButtonPress = this.onCancelButtonPress.bind(this);
+        this.onSearchButtonPress = this.onSearchButtonPress.bind(this);
+        this.onChangeText = this.onChangeText.bind(this);
+    }
+
+    onSearchButtonPress() {
+        const onSearchButtonPress = this.props.onSearchButtonPress;
+        if (onSearchButtonPress && this.state.value) {
+            onSearchButtonPress(this.state.value);
+        }
+    }
+
+    onCancelButtonPress() {
+        const onCancelButtonPress = this.props.onCancelButtonPress;
+        this.setState({
+            isOnFocus: false,
+            value: ''
+        });
+
+        if (onCancelButtonPress) {
```

Comment:
Thoughts on turning this into a defaultProp? You could do
```
defaultProps: {
    ...,
    onCancelButtonPress: () => true
}
```

That would eliminate the need for this logic because it's always going to be there whether it's passed in or not.